### PR TITLE
fix(testing):  add discardPeriodicTasks to be used with fakeAsync

### PR DESCRIPTION
**Pull request by juliemr**

### CODE REVIEW

**Comment by juliemr** [comment]
@vikerman this is the simplest solution - one effect is that it lets the last remaining pending task go after the tick, because it's already been put into the scheduler. To change this, we'd have to modify the fake async spec in `zone.js` to allow public access to the ability to cancel timers. What do you think? I'm not sure what the desired behavior is.

File: modules/@angular/core/test/fake_async_spec.ts [hunk 1]
```diff
@@ -13,13 +13,14 @@ import {
...
   tick,
+  discardPeriodicTasks,
 } from '@angular/core/testing';
 import {TimerWrapper, PromiseWrapper} from '../../router/src/facade/async';
 import {BaseException} from '../../router/src/facade/exceptions';
 import {Parser} from '../../compiler/src/expression_parser/parser';
 
 export function main() {
-  describe('fake async', () => {
+  ddescribe('fake async', () => {
```
Comment (vikerman) [inline modules/@angular/core/test/fake_async_spec.ts]: Please remove ddescribe

File: modules/@angular/core/testing/fake_async.ts [hunk 2]
```diff
@@ -81,6 +81,19 @@ export function tick(millis: number = 0): void {
 }
 
 /**
+ * Discard all remaining periodic tasks.
+ */
+export function discardPeriodicTasks(): void {
+  let zoneSpec = _getFakeAsyncZoneSpec();
+  let pendingTimers = zoneSpec.pendingPeriodicTimers;
+  // console.log(_getFakeAsyncZoneSpec().pendingPeriodicTimers);
+  zoneSpec.pendingPeriodicTimers = [];
```
Comment (vikerman) [inline modules/@angular/core/testing/fake_async.ts]: Can you just do zoneSpec.pendingPeriodicTimer.length = 0 to clear the array? That seems to be standard way.

File: modules/@angular/core/testing/fake_async.ts [hunk 3]
```diff
@@ -81,6 +81,19 @@ export function tick(millis: number = 0): void {
...
+  // pendingTimers.forEach((timer) => {
```
Comment (vikerman) [inline modules/@angular/core/testing/fake_async.ts]: Please remove commented code

**Comment by vikerman** [comment]
There is a test that's failing.

File: modules/@angular/core/test/fake_async_spec.ts (hunk 1 above)
Comment (juliemr) [inline modules/@angular/core/test/fake_async_spec.ts]: done.

File: modules/@angular/core/testing/fake_async.ts (hunk 2 above)
Comment (juliemr) [inline modules/@angular/core/testing/fake_async.ts]: done.

File: modules/@angular/core/testing/fake_async.ts (hunk 3 above)
Comment (juliemr) [inline modules/@angular/core/testing/fake_async.ts]: done.

**Comment by juliemr** [comment]
@vikerman PTAL - the only failure on Travis is also present on master.

**Pull request merged**